adding build process illustration

akoserwal opened the pull request:
After reading the setup steps for the first time, I didn't have an understanding of how the coreos-assembler works. But after watching this talk 

[DevConf talk: Fedora CoreOS Build Tooling by Andrew Jeddeloh](https://www.youtube.com/watch?v=qg3tnxHZu80)

It helps to understand the build process for anyone getting started with the cores-assembler setup process. I think it would be helpful to add this diagram.

lucab commented on README.md:
```diff
@@ -19,6 +19,33 @@ and to keep them as similar as possible:
++-------------------------------------------+                              ||                                  
+                                             \                             \/                                +------------------------+   
+                                              \          +-----------------------------------+               |  Volumne mounted in    |
+                                               \         |  CoreOs Assembler Container       |               |                        |
```
"CoreOS" or `coreos-assembler`.

lucab commented on README.md:
```diff
@@ -19,6 +19,33 @@ and to keep them as similar as possible:
+|                                           |                              || Build on quay.io
++-------------------------------------------+                              ||                                  
+                                             \                             \/                                +------------------------+   
+                                              \          +-----------------------------------+               |  Volumne mounted in    |
```
Typo: s/Volumne/Volume/

akoserwal replied:
@lucab: I'll fix it

lucab commented:
Thanks for the PR! I'm a big fan of AsciiArt diagrams but at the same time they are not the easiest thing to update when things change. Is your diagram in some standard dialect like [mermaid](https://github.com/knsv/mermaid) or [svgbob](https://github.com/ivanceras/svgbob) that we can easily get an SVG from?

dustymabe commented on README.md:
```diff
@@ -19,6 +19,33 @@ and to keep them as similar as possible:
+|         FCOS Config                       |            |                                  |
+|                                           |            |   * Build Script                 |
+|    * rpm-ostree treefile                  |            |   * Dockerfile                   |
+|    * Kickstart                            |            |                                  |
```
I think we removed the kickstart from the config - so probably can remove that.

ajeddeloh replied:
It's currently part of the build scripts.

dustymabe replied:
but it's definitely not part of the FCOS config github repo

akoserwal replied:
@dustymabe @ajeddeloh: should I remove the kickstart from the FOCS config?

dustymabe replied:
I would

akoserwal replied:
@dustymabe: sure, let me remove it.

ajeddeloh commented:
:tada: 

Misc note: It might be useful to have it flow top to bottom instead of left to right; currently it ends up scrolling sideways when previewing the rendered markdown.

akoserwal commented:
@lucab: using mermaid would be a good idea. Here is the exported SVG from mermaid:
<img src="https://gist.githubusercontent.com/akoserwal/fc1d26b0db93b9531e4f1284e563008e/raw/725bba404a804c00e6be3ec7ce17496f9631181e/core-os-build-process.svg?sanitize=true">

dustymabe commented:
> @dustymabe: Is this looks good. I can update my PR with this svg.

looks good to me i might argue that things would need to be changed slightly but maybe I can propose that change in a follow up. Is the input to create the SVG plain text? If not I'd prefer to keep whatever source we use to create the resulting image as plain text so it can be easily updated in git. When you update your commit can you add the source file?

akoserwal commented:
> > @dustymabe: Is this looks good. I can update my PR with this svg.
> 
> looks good to me i might argue that things would need to be changed slightly but maybe I can propose that change in a follow up. Is the input to create the SVG plain text? If not I'd prefer to keep whatever source we use to create the resulting image as plain text so it can be easily updated in git. When you update your commit can you add the source file?

I used [mermaidjs](https://mermaidjs.github.io/) 
this is the script
```
graph LR
 A(<div><h3>FCOS Config</h3><ul><li>rpm-ostree treefile</li><li>Repos</li><li>Misc other bits</li></ul>)  --> C[<h3>coreos-assembler container</h3><ul><li>Fedora Base</li><li>Build Script Installed</li></ul></div>];
D[RPMS] --> C;
E[<h3>coreos-assembler repo </h3> <ul><li>Build Script</li><li>Dockerfile</li></ul>]  --Build on quay.io --> C;
C --> F[Volume mounted in <ul><li>Disk Images</li>ostree commit</li>.</ul>]

classDef green fill:#9f6,stroke:#333,stroke-width:1px, width:300px, padding:30px;
class A green;

```
Copy paste in the live editor: https://mermaidjs.github.io/mermaid-live-editor/

The only challenge with this tool is that it does have an issue with css/style for the elements. when I exported in the svg format. I have to update the style directly in the svg.

dustymabe commented:
interesting.. yeah can we at least get that source file added as part of the commit?

akoserwal commented:
> interesting.. yeah can we at least get that source file added as part of the commit?

I updated my PR with SVG and added another file: build-process-mermaid.md with source script and instructions to update the SVG. Review preview will show that image linking is broken, but it seems to be a bug in GitHub (https://github.com/coreos/coreos-assembler/pull/520/files#diff-04c6e90faac2675aa89e2176d2eec7d8). Once merged it will be linked:https://github.com/akoserwal/coreos-assembler/blob/build-process-info/README.md

sinnykumari commented:
LGTM ,  can you rebase your branch to avoid conflict?
Nice work @akoserwal !

akoserwal commented:
> LGTM , can you rebase your branch to avoid conflict?
> Nice work @akoserwal !

Thanks, Done.
